core/src/main/java/org/apache/iceberg/BaseRewriteFiles.java

```diff
@@ -26,6 +26,9 @@
 class BaseRewriteFiles extends MergingSnapshotProducer<RewriteFiles> implements RewriteFiles {
   private final Set<DataFile> replacedDataFiles = Sets.newHashSet();
   private Long startingSnapshotId = null;
+  private boolean replacesDeleteFiles = false;
```

Comment:
rather than fields, can we check whether "repalcesDataFiles" or other data structures have contents?

Reply:
Do you mean checking the underlying `ManifestFilterManager`, `List<DataFile>` and the map of delete files? I thought about this but was initially reluctant to touch `ManifestFilterManager` (4 different ways to delete files).

That said, these boolean flags are a bit odd. Now I am thinking it makes sense to spend a bit more time and check the underlying structures. @RussellSpitzer, is this what you meant?

Reply:
Yep, sorry if I was unclear. It's just weird to keep this as state flags when they are just proxy variables for the actual state of containers we have access to.